ql/src/java/org/apache/hadoop/hive/ql/exec/tez/YarnQueueHelper.java

```diff
@@ -94,19 +94,15 @@ public YarnQueueHelper(HiveConf conf) {
   }
 
   public void checkQueueAccess(
-      String queueName, String userName) throws IOException, InterruptedException {
+      String queueName, String userName) throws IOException {
     UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
     try {
       ugi.doAs((PrivilegedExceptionAction<Void>) () -> {
         checkQueueAccessInternal(queueName, userName);
         return null;
       });
-    } finally {
-      try {
-        FileSystem.closeAllForUGI(ugi);
-      } catch (IOException exception) {
-        LOG.error("Could not clean up file-system handles for UGI: " + ugi, exception);
-      }
```

Comment:
Who closes the FileSystem now? If there is a FileSystem created per UGI and we don't clean it up, can that lead to OOM issues?

Reply:
@ayushtkn There is no need to call FileSystem.closeAllForUGI(ugi) here, the close is handled elsewhere and worked fine earlier. When I added the `ugi.doAs(...` call in https://github.com/apache/hive/pull/3402, I accidentally added the finally block as well, which is not required here at all.

Reply:
ohhhk, @veghlaci05 is this reproducible via a UT?

Reply:
Unorftunately not, I was able to repro it only using a prepared cluster.

Reply:
cool, go ahead